src/ucp/core/ucp_context.c

```diff
@@ -574,8 +574,18 @@ static ucs_status_t ucp_fill_config(ucp_context_h context,
     context->config.request.size    = params->request_size;
     context->config.request.init    = params->request_init;
     context->config.request.cleanup = params->request_cleanup;
+#if ENABLE_UCP_THREAD
+    context->config.thread_mode     = params->thread_mode;
+#else
+    context->config.thread_mode     = UCS_THREAD_MODE_SINGLE;
+#endif
     context->config.ext             = config->ctx;
 
+    if (context->config.thread_mode == UCS_THREAD_MODE_MULTI) {
+        /* initializing thread mutex on this context */
+        pthread_mutex_init(&(context->config.thread_mutex), NULL);
```

Comment:
the mutex is not part of the "configuration", it's part of the context itself, this is not the right place to initialize the mutex